doc/source/guidelines/optional_packages.rst

```diff
@@ -0,0 +1,248 @@
+.. _optional_packages:
+
+Handling Optional Packages
+==========================
+
+The ``setuptools``, ``flit``, and ``poetry`` build systems all allow you to
+declare dependencies that only get installed under specific circumstances. This
+makes it convenient to declare dependencies for ancillary functions such as
+"plotting", "tests", or "docs". This allows you to pragmatically integrate
+optional dependencies which can be installed as optional requirements rather
+than individual packages.
+
+You may want to have optional packages for your pyansys library for a variety
+of reasons, including:
+
+- **Not all users will want to use the feature.** - For example, you might want
+  to make the usage of `matplotlib <https://matplotlib.org/`_ or `pyvista
+  <https://docs.pyvista.org/>`_ if you expect your pyansys library to be used
+  primarily for headless scripting rather than visualization.
+- **Not all users can install the optional package.** - For certain less popular
+  or obscure environments, some binary wheels may not be available or
+  compatible with the user's environment. For example, if they're using CentOS
+  6.9 and need a ``manylinux1`` but the package only supports ``manylinux2014``
+  (CentOS 7+) or newer, their environment wouldn't be able to run the pyansys
+  library.
+- **Reducing Dependency Bloat** - Removing the package as a "required"
+  dependency reduces the number of packages installed at install-time, speeding
+  up the install and reducing the possibility of dependency conflicts. The
+  trade-off here is any user who wishes to access features which require the
+  optional package will have to install it piecemeal.
+
+If you choose to implement optional packages for your pyansys library, here are
+some helpful best-practices to follow.
+
+
+Implementing Optional Packages in the Build System
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Here's how to implement and use optional requirements for the three most
+popular build systems.
+
+.. tabs::
+
+   .. tab:: poetry
+
+      .. code-block::
+
+         [tool.poetry.extras]
```

Comment:
AFAIK this is insufficient; ``poetry`` **additionally** needs the dependency to be declared in the ``[tool.poetry.dependencies]`` section, e.g.

```toml
matplotlib = {version = "^3.5", optional = true}
```